e2e/wdio/platform/tests/wizard-generator.e2e-spec.ts

```diff
@@ -0,0 +1,545 @@
+            ? click(selector + nextStepBtn2)
+            : click(selector + nextStepBtn);
+        expect(getElementClass(selector + step, 1)).toContain('current', 'you not moved to second step');
+        expect(getElementClass(selector + step, 0)).toContain('completed', 'first step is not completed');
+        click(selector + stepContainer);
+        expect(getElementClass(selector + step, 0)).toContain(
+            'current',
+            'you did not get back to first step by click on step tab'
+        );
+    }
+
+    function checkValidationRequiredFields(selector: string): void {
+        scrollIntoView(selector);
+        selector === externalNavigationExample || selector === customizableGeneratorExample || selector === dialog
+            ? click(selector + nextStepBtn2)
+            : click(selector + nextStepBtn);
+        if (selector !== summaryObjectsExample) {
+            expect(getElementClass(selector + selectControl)).toContain('error', 'error is not appeared');
+            click(selector + select);
+            click(listItem);
+        }
+        if (selector === summaryObjectsExample) {
+            expect(getElementClass(selector + checkbox)).toContain('error', 'error is not appeared');
+            click(checkboxLabel);
+        }
+        if (
+            selector !== externalNavigationExample &&
+            selector !== customizableGeneratorExample &&
+            selector !== dialog
+        ) {
+            click(selector + nextStepBtn);
+            click(selector + nextStepBtn, 1);
+        }
+        if (
+            selector === externalNavigationExample ||
+            selector === customizableGeneratorExample ||
+            selector === dialog
+        ) {
+            click(selector + nextStepBtn2);
+            click(selector + nextStepBtn2);
+        }
+        expect(getElementClass(selector + input)).toContain('error', 'error is not appeared');
+        expect(getElementClass(selector + input, 1)).toContain('error', 'error is not appeared');
+        setValue(selector + input, name);
+        setValue(selector + input, firstAdress, 1);
+        if (
+            selector !== externalNavigationExample &&
+            selector !== customizableGeneratorExample &&
+            selector !== dialog
+        ) {
+            click(selector + nextStepBtn, 1);
+            click(selector + nextStepBtn, 2);
+            expect(getElementClass(selector + input, 3)).toContain('error', 'error is not appeared');
+        }
+        if (
+            selector === externalNavigationExample ||
+            selector === customizableGeneratorExample ||
+            selector === dialog
+        ) {
+            click(selector + nextStepBtn2);
+            click(selector + nextStepBtn2);
+            expect(getElementClass(selector + input)).toContain('error', 'error is not appeared');
+        }
+    }
+
+    function checkBasicWay(selector: string): void {
```

Comment:
better to use many simple functions than 1 large complex one. Here you are asserting each step, is step 1 fails the erst of the function won't run. I suggest doing like this:
completeStepOne(); -- where you simply perform the actions, no assertions
completeStepTwo()
..completeStep3..
..summary..

then write test cases like this:
describe(x example .....
it('should check first step'),....
completeStepOne(..);   --- again, this method is just actions
expect(....)...;   --- assertions after actions
}
it(step 2)..
completStepOne()
click(nextStepButton);
completeStepTwo()
expect...

this way we will instantly know which step has failed, functions will (hopefully) be easier to read, and only doing assertions for what we are checking and all assertions will be at the end so if they fail there won't be skipped code.

Reply:
also, you may want to look into using a switch case instead of 'if' here